charmhelpers/contrib/network/files/check_sriov_numvfs.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""
+Check to verify the number of virtual functions (VFs) active on network
+interfaces. Based on /sys/class/net/<interface>/device/sriov_numvfs and
+/sys/class/net/<interface>/device/sriov_totalvfs.
+
+    usage: check_sriov_numvfs.py [-h] sriov_numvfs [sriov_numvfs ...]
+
+    Check sriov numvfs configuration
+
+    positional arguments:
+    sriov_numvfs  format: <interface>:<numvfs>
+
+For each interfaces:numvfs pair given it verifies that:
+    1. VFs are enabled
+    2. VFs are configured are the same as specified in the check parameter
+    3. VFs configured do not exceed the maximum available on the interface
+
+Example: ./check_sriov_numvfs.py ens3f0:32 ens3f1:32 ens6f0:32 ens6f1:32
+"""
+
+import argparse
+import os.path
+import sys
+
+
+DEVICE_TEMPLATE = "/sys/class/net/{0}"
+SRIOV_NUMVFS_TEMPLATE = "/sys/class/net/{0}/device/sriov_numvfs"
+SRIOV_TOTALVFS_TEMPLATE = "/sys/class/net/{0}/device/sriov_totalvfs"
+
+
+def get_interface_setting(file):
+    """ Return the value content of a setting file as int """
+    with open(file) as f:
+        value = f.read()
+    return int(value)
+
+
+def check_interface_numvfs(iface, numvfs):
+    """ Check SRIOV numvfs config """
+    sriov_numvfs_path = SRIOV_NUMVFS_TEMPLATE.format(iface)
+    sriov_totalvfs_path = SRIOV_TOTALVFS_TEMPLATE.format(iface)
+    msg = []
+    if os.path.exists(DEVICE_TEMPLATE.format(iface)):
```

Comment:
While this works, the structure feels a little unwieldly.  I would personally suggest refactoring it, so that rather than getting to 3-4 levels of indentation deep, you instead return early where possible.  e.g. if the device template path doesn't exist, just return msg at that point and then the rest of the function can be de-dented by a level?

Reply:
Improved, should be more straightforward now